openssl/src/ssl/test.rs

```diff
@@ -1536,3 +1536,39 @@ fn stateless() {
     send(client_stream.get_mut(), server_stream.get_mut());
     hs(server_stream.handshake()).unwrap();
 }
+
+#[cfg(not(osslconf = "OPENSSL_NO_PSK"))]
+#[test]
+fn psk_ciphers() {
+    const CIPHER: &'static str = "PSK-AES128-CBC-SHA";
+    const PSK: &[u8] = b"thisisaverysecurekey";
+    const CLIENT_IDENT: &[u8] = b"thisisaclient";
+
+    let listener = TcpListener::bind("127.0.0.1:0").unwrap();
+    let port = listener.local_addr().unwrap().port();
+
+    thread::spawn(move || {
+        let stream = listener.accept().unwrap().0;
+        let mut ctx = SslContext::builder(SslMethod::tls()).unwrap();
+        ctx.set_cipher_list(CIPHER).unwrap();
+        ctx.set_psk_server_callback(move |_, identity, psk| {
```

Comment:
Could you add a bit of code to ensure that these callbacks are being called? I'll look a bit like this test, for example: https://github.com/sfackler/rust-openssl/blob/4f3d72d98c527f571ac50fcb41317bc44d0fbfda/openssl/src/ssl/test.rs#L1026-L1063

Looks good other than this one last thing!